File: java/oop/CollegeApp/src/com/techlabs/college/Professor.java
package com.techlabs.college;

public class Professor extends Person implements SalariedEmployee {
	private static final double salaryPerHour;
	private int totalWorkingHours;
	private double totalSalary;
	
	static{
		salaryPerHour=5000;
	}
	
	public Professor(String name,String address,String dob,int id,Branch branch,int workinghours){
		super(name,address,dob,id,branch);	
		this.totalWorkingHours=workinghours;
	}
	
	public double getSalary(){
		return this.totalSalary;
	}
	
	public double calculateSalary(){
		
		this.totalSalary= this.totalWorkingHours*salaryPerHour;
		return totalSalary;
	}
	

}

File: dotNet/csharp/CollegeApp/CollegeApp/Professor.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace CollegeApp
{
    class Professor :Person,SalariedEmployee
    {
        private static double _salaryPerHour;
        private int _totalWorkingHours;
        private double _totalSalary;

        static Professor()
        {
            _salaryPerHour = 500;
        }

        public Professor(string name,string address,string dob,int id,Branch branch,int workinghours):base(name,address,dob,id,branch)
        {
            this._totalWorkingHours = workinghours;
        }

        public double TotalSalary
        {
            get
            {
                return _totalSalary;
            }
        }

        public double CalculateSalary()
        {
            this._totalSalary = this._totalWorkingHours * Professor._salaryPerHour;
            return _totalSalary;
        }
    }
}
